pkg/config/duration.go

```diff
@@ -27,12 +27,51 @@ func MustNewDuration(d time.Duration) *Duration {
 	return &rv
 }
 
+func MakeDuration(d time.Duration) (Duration, error) {
+	if d < time.Duration(0) {
+		return Duration{}, fmt.Errorf("cannot make negative time duration: %s", d)
+	}
+	return Duration{d: d}, nil
+}
+
+func ParseDuration(s string) (Duration, error) {
+	d, err := time.ParseDuration(s)
+	if err != nil {
+		return Duration{}, err
+	}
+
+	return MakeDuration(d)
+}
+
+func MustMakeDuration(d time.Duration) Duration {
+	rv, err := MakeDuration(d)
+	if err != nil {
+		panic(err)
+	}
+	return rv
+}
```

Comment:
This is basically the same as `MustNewDuration`, and callers could just call `*MustNewDuration(d)` to get the same result.
```suggestion
```
I don't love that `New` and `MustNew` are not aligned, but I think it would still be simpler to keep this API small by just using those directly in these cases.

Reply:
Agreed, I'll make this update as well.